src/core/datatypes/math/OrderedEventTimes.cpp

```diff
@@ -0,0 +1,166 @@
+/*
+ * EventTime.cpp
+ *
+ *  Created on: Apr 9, 2020
+ *      Author: mike
+ */
+
+#include <ostream>
+
+#include "OrderedEventTimes.h"
+#include "TypedDagNode.h"
+
+namespace RevBayesCore {
+
+OrderedEventTimes::OrderedEventTimes()
+{
+}
+
+OrderedEventTimes::~OrderedEventTimes()
+{
+}
+
+OrderedEventTimes* OrderedEventTimes::clone(void) const
+{
+    return new OrderedEventTimes( *this );
+}
+
+bool OrderedEventTimes::operator==(const OrderedEventTimes &oet) const
+{
+	return event_times == oet.event_times;
+}
+
+bool OrderedEventTimes::operator!=(const OrderedEventTimes &oet) const
+{
+	return event_times != oet.event_times;
+}
+
+bool OrderedEventTimes::operator<(const  OrderedEventTimes &oet) const
+{
+	return false;
```

Comment:
Is this something where you have to implement operator<( ), because its a pure virtual function?
It might be better to throw an exception, to prevent (say) sorting such objects or placing them into a std::map.

Reply:
Alternatively, just calling `abort()` is good for debugging if you think this should never be seen by the end user.

Reply:
Actually we should step back: I think ordered event times (and everything else that depends on it, like Markov event distributions) should probably be taken out of this merge. They're totally experimental.